plugins/inputs/prometheus/metrics_type_handler.go

```diff
@@ -89,7 +89,7 @@ func (t *metadataServiceImpl) Get(job, instance string) (metadataCache, error) {
 
 	// from the same targetGroup, instance is not going to be duplicated
 	for _, target := range targetGroup {
-		if target.Labels().Get(savedScrapeInstanceLabel) == instance {
+		if target.DiscoveredLabels().Get(savedScrapeInstanceLabel) == instance || target.DiscoveredLabels().Get(scrapeInstanceLabel) == instance {
```

Comment:
The cache now uses __address__ keeping cwagent_saved_scrape_instance for older versions of k8s